arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift

```diff
@@ -0,0 +1,323 @@
+// Copyright 2020 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import AVFoundation
+import ArcGIS
+import ArcGISToolkit
+
+// MARK: - Navigate the route
+
+class NavigateARNavigatorViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The label to display route-planning status.
+    @IBOutlet weak var statusLabel: UILabel!
+    /// The bar button to calibrate navigation heading.
+    @IBOutlet weak var calibrateButtonItem: UIBarButtonItem!
+    /// The bar button to start navigation.
+    @IBOutlet weak var startButtonItem: UIBarButtonItem!
+    /// The `ArcGISARView` managed by the view controller.
+    @IBOutlet weak var arView: ArcGISARView! {
+        didSet {
+            arView.sceneView.scene = makeScene()
+            arView.sceneView.graphicsOverlays.add(makeRouteOverlay())
+            // Turn the space and atmosphere effects on for an immersive experience.
+            arView.sceneView.spaceEffect = .transparent
+            arView.sceneView.atmosphereEffect = .none
+            arView.locationDataSource = AGSCLLocationDataSource()
+        }
+    }
+    
+    // MARK: Instance properties
+    
+    /// The route result solved by the route task, received from route planner.
+    var routeResult: AGSRouteResult!
+    /// The route task that solves the route using the online routing service, received from route planner.
+    var routeTask: AGSRouteTask!
+    /// The parameters for route task to solve a route, received from route planner.
+    var routeParameters: AGSRouteParameters!
+    
+    /// A boolean that indicates if the heading is being calibrated.
+    var isCalibrating = false {
```

Comment:
Instead of having this be a stored property, consider making it computed. Something like the following:

```swift
var isCalibrating: Bool {
    presentedViewController is NavigateARCalibrationViewController
}
```

Then just adjust the opacity when the calibration view controller is presented and change it back when the calibration view controller is dismissed.